src/level_set/FESurfaceDistanceEvaluator.cpp

```diff
@@ -877,21 +1143,174 @@ FESurfaceDistanceEvaluator::getClosestPoint3D(const IBTK::Vector3d& P,
+            const double nbr_e1_norm = nbr_e1.norm();
+            const double angle = acos(nbr_e0.dot(nbr_e1) / (nbr_e0_norm * nbr_e1_norm));
+            angle_weighted_normal += (angle * d_elem_face_normal[nbr_elem]);
+        }
+        angle_weighted_normal /= angle_weighted_normal.norm();
+        break;
+    case VERTEX_1:
+        elem_set = d_node_to_elem[elem->node_ptr(1)];
+        for (const auto& itr : elem_set)
+        {
+            Elem* nbr_elem = itr;
+            const libMesh::Point& nbr_n0 = nbr_elem->point(0);
+            const libMesh::Point& nbr_n1 = nbr_elem->point(1);
+            const libMesh::Point& nbr_n2 = nbr_elem->point(2);
+            if (nbr_elem->node_ptr(0) == elem->node_ptr(1))
+            {
+                nbr_e0 << MAKE_EDGE(nbr_n0, nbr_n1);
+                nbr_e1 << MAKE_EDGE(nbr_n0, nbr_n2);
+            }
+            else if (nbr_elem->node_ptr(1) == elem->node_ptr(1))
+            {
+                nbr_e0 << MAKE_EDGE(nbr_n1, nbr_n0);
+                nbr_e1 << MAKE_EDGE(nbr_n1, nbr_n2);
+            }
+            else if (nbr_elem->node_ptr(2) == elem->node_ptr(1))
+            {
+                nbr_e0 << MAKE_EDGE(nbr_n2, nbr_n0);
+                nbr_e1 << MAKE_EDGE(nbr_n2, nbr_n1);
+            }
+            const double nbr_e0_norm = nbr_e0.norm();
+            const double nbr_e1_norm = nbr_e1.norm();
+            const double angle = acos(nbr_e0.dot(nbr_e1) / (nbr_e0_norm * nbr_e1_norm));
+            angle_weighted_normal += (angle * d_elem_face_normal[nbr_elem]);
+        }
+        angle_weighted_normal /= angle_weighted_normal.norm();
+        break;
+    case VERTEX_2:
+        elem_set = d_node_to_elem[elem->node_ptr(2)];
+        for (const auto& itr : elem_set)
+        {
+            Elem* nbr_elem = itr;
+            const libMesh::Point& nbr_n0 = nbr_elem->point(0);
+            const libMesh::Point& nbr_n1 = nbr_elem->point(1);
+            const libMesh::Point& nbr_n2 = nbr_elem->point(2);
+            if (nbr_elem->node_ptr(0) == elem->node_ptr(2))
+            {
+                nbr_e0 << MAKE_EDGE(nbr_n0, nbr_n1);
+                nbr_e1 << MAKE_EDGE(nbr_n0, nbr_n2);
+            }
+            else if (nbr_elem->node_ptr(1) == elem->node_ptr(2))
+            {
+                nbr_e0 << MAKE_EDGE(nbr_n1, nbr_n0);
+                nbr_e1 << MAKE_EDGE(nbr_n1, nbr_n2);
+            }
+            else if (nbr_elem->node_ptr(2) == elem->node_ptr(2))
+            {
+                nbr_e0 << MAKE_EDGE(nbr_n2, nbr_n0);
+                nbr_e1 << MAKE_EDGE(nbr_n2, nbr_n1);
+            }
+            const double nbr_e0_norm = nbr_e0.norm();
+            const double nbr_e1_norm = nbr_e1.norm();
+            const double angle = acos(nbr_e0.dot(nbr_e1) / (nbr_e0_norm * nbr_e1_norm));
+            angle_weighted_normal += (angle * d_elem_face_normal[nbr_elem]);
+        }
+        angle_weighted_normal /= angle_weighted_normal.norm();
+        break;
```

Comment:
```suggestion
        break;
    case UNKNOWN_FEATURE:
    default:
        TBOX_ERROR("Unknown geometric feature");
```

Reply:
There might be a better error message, but its quite risky to not cover all cases and have a `default:` at the end.

Reply:
`switch` statements should always have a `default` case.

Reply:
I added the default case.

Reply:
We have a safety catch of `TBOX_ASSERT(feature != UNKNOWN_FEATURE)` upstream of `switch`.

Reply:
Sure, but its still a compiler warning to not define all switch cases.

Reply:
I see. Now all 8 switch cases are defined, so hopefully there are no more compiler warnings for this.

Reply:
I think it is good practice to always include a `default` case. Reasons include addressing the situation in which items are added to the `enum` at some point in the future, and in which case statements are _removed_ (either accidentally or on purpose). It is easier to debug these things if there is always a defined default behavior other than silently passing through the `switch`.

Reply:
"It's not paranoia if they are actually out to get you"

Reply:
The template function made it look even uglier than the original in-place substitution of `n0-n1`. Hence, I had recommended using a macro. 

But there may be other advantages to switching to a function than a macro.